notebooks/config.ipynb

```diff
@@ -964,12 +971,831 @@
+      " |  Methods inherited from collections.abc.Mapping:\n",
+      " |\n",
+      " |  __eq__(self, other)\n",
+      " |      Return self==value.\n",
+      " |\n",
+      " |  items(self)\n",
+      " |      D.items() -> a set-like object providing a view on D's items\n",
+      " |\n",
+      " |  keys(self)\n",
+      " |      D.keys() -> a set-like object providing a view on D's keys\n",
+      " |\n",
+      " |  values(self)\n",
+      " |      D.values() -> an object providing a view on D's values\n",
+      " |\n",
+      " |  ----------------------------------------------------------------------\n",
+      " |  Data and other attributes inherited from collections.abc.Mapping:\n",
+      " |\n",
+      " |  __hash__ = None\n",
+      " |\n",
+      " |  __reversed__ = None\n",
+      " |\n",
+      " |  ----------------------------------------------------------------------\n",
+      " |  Class methods inherited from collections.abc.Collection:\n",
+      " |\n",
+      " |  __subclasshook__(C)\n",
+      " |      Abstract classes can override this to customize issubclass().\n",
+      " |\n",
+      " |      This is invoked early on by abc.ABCMeta.__subclasscheck__().\n",
+      " |      It should return True, False or NotImplemented.  If it returns\n",
+      " |      NotImplemented, the normal algorithm is used.  Otherwise, it\n",
+      " |      overrides the normal algorithm (and the outcome is cached).\n",
+      " |\n",
+      " |  ----------------------------------------------------------------------\n",
+      " |  Class methods inherited from collections.abc.Iterable:\n",
+      " |\n",
+      " |  __class_getitem__ = GenericAlias(...)\n",
+      " |      Represent a PEP 585 generic type\n",
+      " |\n",
+      " |      E.g. for t = list[int], t.__origin__ is list and t.__args__ is (int,).\n",
+      "\n"
+     ]
+    }
+   ],
+   "source": [
+    "help(config.INIConfig)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "91a793e9-d1d1-41bb-84fd-c0c279606473",
+   "metadata": {},
+   "source": [
+    "An object can be initialized by providing a config either as a Python `dict` or a <a href=\"https://docs.python.org/3/library/pathlib.html#pathlib.Path\">Path</a> to the file.\n",
+    "<!--cell 74-->"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 38,
+   "id": "0e8d6c8a-cc3c-49d7-9b97-32187cd5f754",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "[fruit count]\n",
+      "apples = 3\n",
+      "grapes = {{ grape_count }}\n",
+      "kiwis = 2\n"
+     ]
+    }
+   ],
+   "source": [
+    "fruits = config.INIConfig(\n",
+    "    config=Path('fixtures/config/fruit-config.ini')\n",
+    ")\n",
+    "print(fruits)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1a50a362-d063-4ae1-9b6a-d98d0bf84c4f",
+   "metadata": {},
+   "source": [
+    "### Comparing Config Objects\n",
```

Comment:
This is interesting. I'd expect that we'd want to explicitly mention that all the config objects have this capability and that this is an alternative to using the `compare()` tool above from the API, which only takes file paths as input.

It may also be worth mentioning that while a dictionary doesn't inherently have a file format (INI, sh, etc.), the internals of UW Tools will specify how deep these formats can be. Those format limitations are not applied to this function (as best I can tell...please correct me if I read that wrong) so all format objects will give you the same behavior on any two dictionaries.

Reply:
I like these suggestions and added the information here to the markdown cell. The method used here doesn't have the depth limitations corresponding to each format applied, as you said, but it does assume a section/key/value structure, and I indicate this in the updated markdown. 

I think this might create a problem for some formats, especially Shell configs, which would not be able to compare any other dictionary to one that a SHConfig object contains.